python/pyhie/allspark/model_loader.py

```
@@ -908,6 +908,23 @@ def load_model(self, override_data_type=None, direct_load=False, load_format="au
+        if not direct_load:
+            index_file = os.path.join(self.hf_model_path, 'model.safetensors.index.json')
+            if os.path.exists(index_file):
+                try:
+                    with open(index_file) as f:
+                        idx = json.load(f)
+                    if any(k.endswith('.weight_scale_inv')
+                           for k in idx.get('weight_map', {})):
+                        print("[INFO] FP8 model detected, auto-enabling direct_load=True")
+                        direct_load = True
+                except Exception:
+                    pass
```

Ruff (0.15.1)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)